packages/nep141-erc20/src/natural-erc20/sendToNear/index.js

```diff
@@ -503,8 +506,32 @@ async function checkSync (transfer) {
   const eventEmittedAt = lockReceipt.blockNumber
   const syncedTo = await ethOnNearSyncHeight()
   const completedConfirmations = Math.max(0, syncedTo - eventEmittedAt)
+  let proof
+
+  if (completedConfirmations > transfer.neededConfirmations) {
+    // Check if relayer already minted
+    proof = await findProof(lockReceipt.transactionHash)
+    const nearAccount = await getNearAccount()
+    const proofAlreadyUsed = await nearAccount.viewFunction(
+      process.env.nearTokenFactoryAccount,
+      'is_used_proof',
+      Buffer.from(proof)
+    )
+    if (proofAlreadyUsed) {
+      // TODO: find the event relayer tx hash
+      return {
+        ...transfer,
+        completedStep: MINT,
+        completedConfirmations,
+        status: status.COMPLETE,
+        errors: [...transfer.errors, 'Transfer already finalized.']
+        // mintHashes: [...transfer.mintHashes, txHash]
+      }
+    }
+  }
 
-  if (completedConfirmations < transfer.neededConfirmations) {
+  if (completedConfirmations < transfer.neededConfirmations + Number(process.env.nearEventRelayerMargin)) {
```

Comment:
else if

Reply:
This should be checked also for the case where the transfer is confirmed but not yet finalized by the event relayer.